scripts-available/CDB_PG_Federated_Tables.sql

```diff
@@ -0,0 +1,299 @@
+        RAISE EXCEPTION 'non geometry column "%"', webmercator_column;
+    END IF;
+
+    -- Get a list of columns excluding the id, geom and the_geom_webmercator
+    SELECT ARRAY(
+        SELECT quote_ident(c) FROM @extschema@.__ft_getcolumns(src_table) AS c
+        WHERE c NOT IN (id_column, geom_column, webmercator_column, 'cartodb_id', 'the_geom', 'the_geom_webmercator')
+    ) INTO rest_of_cols;
+
+    -- Figure out whether a ST_Transform to 4326 is needed or not
+    IF @postgisschema@.Find_SRID(fdw_objects_name::varchar, table_name::varchar, geom_column::varchar) = 4326
+    THEN
+        geom_expression := format('t.%I AS the_geom', geom_column);
+    ELSE
+        geom_expression := format('@postgisschema@.ST_Transform(t.%I, 4326) AS the_geom', geom_column);
+    END IF;
+
+    -- Figure out whether a ST_Transform to 3857 is needed or not
+    IF Find_SRID(fdw_objects_name::varchar, table_name::varchar, webmercator_column::varchar) = 3857
+    THEN
+        webmercator_expression := format('t.%I AS the_geom_webmercator', webmercator_column);
+    ELSE
+        webmercator_expression := format('@postgisschema@.ST_Transform(t.%I, 3857) AS the_geom_webmercator', webmercator_column);
+    END IF;
+
+    -- Create a view with homogeneous CDB fields
+    EXECUTE format(
+        'CREATE OR REPLACE VIEW %1$I AS
+            SELECT
+                t.%2$I AS cartodb_id,
+                %3$s,
+                %4$s,
+                %5$s
+            FROM %6$s t',
+        table_name,
+        id_column,
+        geom_expression,
+        webmercator_expression,
+        array_to_string(rest_of_cols, ','),
+        src_table
+    );
+
+    -- Grant perms to the view
+    EXECUTE format('GRANT SELECT ON %I TO %s', table_name, fdw_objects_name);
+END
+$$
+LANGUAGE PLPGSQL VOLATILE PARALLEL UNSAFE;
+
+
+CREATE OR REPLACE FUNCTION @extschema@.CDB_SetUp_PG_Federated_Table(
+    server_alias name,
+    schema_name name,
+    table_name name,
+    id_column name,
+    geom_column name
+)
+RETURNS void
+AS $$
+    SELECT @extschema@.CDB_SetUp_PG_Federated_Table(
+        server_alias,
+        schema_name,
+        table_name,
+        id_column,
+        geom_column,
+        geom_column
+    );
+$$
+LANGUAGE SQL VOLATILE PARALLEL UNSAFE;
+
+
+CREATE OR REPLACE FUNCTION @extschema@.CDB_SetUp_PG_Federated_Table(
+    server_alias name,
+    schema_name name,
+    table_name name,
+    id_column name
+)
+RETURNS void
+AS $$
+DECLARE
+    fdw_objects_name NAME := @extschema@.__CDB_User_FDW_Object_Names(server_alias);
+    src_table REGCLASS;
+    rest_of_cols TEXT[];
+BEGIN
+    -- Import the foreign table
+    PERFORM @extschema@.CDB_SetUp_User_PG_FDW_Table(server_alias, schema_name, table_name);
+    src_table := format('%s.%s', fdw_objects_name, table_name);
+
+    -- Check id_column is numeric
+    IF NOT @extschema@.__ft_is_numeric(src_table, id_column) THEN
+        RAISE EXCEPTION 'non integer id_column "%"', id_colun;
```

Comment:
The id should be `id_column`, as it is it's not outputting the correct name. This is something I'd expect the tests had covered, instead I was getting a super odd error message.